File: codegen/results/v002/gemma4_26b_nothink/20260411_180639/python/008_prime_numbers/solution.py
import sys

def get_primes(n):
    """
    Returns a list of all prime numbers up to and including n 
    using the Sieve of Eratosthenes.
    """
    if n < 2:
        return []
    
    # Create a boolean array "is_prime[0..n]" and initialize
    # all entries it as True. A value in is_prime[i] will
    # finally be False if i is Not a prime, else True.
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False
    
    p = 2
    while (p * p <= n):
        # If is_prime[p] is not changed, then it is a prime
        if is_prime[p]:
            # Update all multiples of p starting from p*p
            for i in range(p * p, n + 1, p):
                is_prime[i] = False
        p += 1
    
    # Collect all indices that are still marked as True
    return [p for p in range(n + 1) if is_prime[p]]

def main():
    # Check if an argument was provided
    if len(sys.argv) != 2:
        # Depending on requirements, you might want to exit or show usage
        # Here we simply exit as per the instruction to "Accept a single integer"
        return

    try:
        # Convert the command line argument to an integer
        n = int(sys.argv[1])
        
        # Generate the primes
        primes = get_primes(n)
        
        # Print each prime on a new line
        for prime in primes:
            print(prime)
            
    except ValueError:
        # If the argument is not an integer, do nothing
        pass

if __name__ == "__main__":
    main()

File: codegen/results/v002/gemma4_26b_nothink/20260411_180639/typescript/008_prime_numbers/solution.ts
/**
 * This program takes a single integer N from the command line 
 * and prints all prime numbers up to and including N, one per line.
 * It uses the Sieve of Eratosthenes algorithm for efficiency.
 */

import process from 'node:process';

function solve(): void {
    // Get the command line argument
    const arg = process.argv[2];

    // Validate that an argument is provided
    if (arg === undefined) {
        return;
    }

    // Parse the argument to an integer
    const n = parseInt(arg, 10);

    // If N is not a valid number or less than 2, there are no primes to print
    if (isNaN(n) || n < 2) {
        return;
    }

    // Use Sieve of Eratosthenes to find primes up to N
    // Create a boolean array "isPrime[0..n]" and initialize
    // all entries it as true. A value in isPrime[i] will
    // finally be false if i is Not a prime, else true.
    const isPrime: boolean[] = new Array(n + 1).fill(true);
    isPrime[0] = false;
    isPrime[1] = false;

    for (let p = 2; p * p <= n; p++) {
        // If isPrime[p] is not changed, then it is a prime
        if (isPrime[p]) {
            // Update all multiples of p starting from p*p
            for (let i = p * p; i <= n; i += p) {
                isPrime[i] = false;
            }
        }
    }

    // Collect and print all primes
    let output = '';
    for (let p = 2; p <= n; p++) {
        if (isPrime[p]) {
            output += p + '\n';
        }
    }

    // Write to stdout
    process.stdout.write(output);
}

// Execute the solver
solve();